src/components/Radio/index.test.tsx

```diff
@@ -0,0 +1,283 @@
+  })
+
+  describe('when it is clicked', () => {
+    describe('and it has not been checked before', () => {
+      it('fires an on change event ', () => {
+        const handleChange = jest.fn(event => event.target.value)
+        const valueMock = 'test'
+        const ref = createRef<HTMLInputElement>()
+
+        render(
+          <Radio
+            ref={ref}
+            onChange={handleChange}
+            value={valueMock}
+            checked={false}
+            id="test"
+            name="test"
+          />
+        )
+
+        userEvent.click(screen.getByRole('radio'))
+
+        expect(handleChange).toBeCalledTimes(1)
+        expect(handleChange).toHaveBeenCalledWith(
+          expect.objectContaining({
+            target: expect.objectContaining({
+              value: valueMock,
+            }),
+          })
+        )
+      })
+    })
+
+    describe('and it is already checked', () => {
+      it('does not fire an on change event', () => {
+        const handleChange = jest.fn(event => event.target.value)
+        const ref = createRef<HTMLInputElement>()
+
+        render(
+          <Radio
+            ref={ref}
+            onChange={handleChange}
+            value="test"
+            checked={true}
+            id="test"
+            name="test"
+          />
+        )
+
+        userEvent.click(ref.current)
+
+        expect(handleChange).toBeCalledTimes(0)
+      })
+    })
+
+    describe('and it is disabled', () => {
+      it('does not fire an on change event', () => {
+        const handleChange = jest.fn(event => event.target.value)
+        const ref = createRef<HTMLInputElement>()
+
+        render(
+          <Radio
+            ref={ref}
+            onChange={handleChange}
+            value="test"
+            checked={false}
+            id="test"
+            name="test"
+            disabled
+          />
+        )
+
+        userEvent.click(ref.current)
+
+        expect(handleChange).toBeCalledTimes(0)
+      })
+    })
+  })
+})
+
+describe('RadioWithLabel', () => {
+  it('renders without crashing', () => {
+    render(
+      <RadioWithLabel
+        onChange={() => null}
+        value="on"
+        checked
+        id="test"
+        name="test"
+        label="Radio 1"
+      />
+    )
+
+    expect(screen.getByText('Radio 1')).toBeInTheDocument()
+    expect(
+      screen.getByRole('presentation', {
+        name: /checkmark/i,
+      })
+    ).toBeInTheDocument()
+  })
+
+  describe('on click', () => {
+    it('fires an on change event when not checked before', () => {
+      const handleChange = jest.fn(event => event.target.value)
+      const valueMock = 'test'
+      const ref = createRef<HTMLInputElement>()
+
+      render(
+        <RadioWithLabel
+          ref={ref}
+          onChange={handleChange}
+          value={valueMock}
+          checked={false}
+          id="test"
+          name="test"
+          label="test"
+        />
+      )
+
+      userEvent.click(ref.current)
```

Comment:
I think it is beter to click the label, because you already tested this on the radio without label.